backend/api/views.py

```diff
@@ -1,77 +1,74 @@
-from rest_framework.decorators import api_view
 from rest_framework.response import Response
 from rest_framework import status
+from rest_framework.permissions import IsAuthenticated, AllowAny
+from rest_framework.views import APIView
 from .serializers import FiboUserSerializer
 from .models import FiboUser
-from django.contrib.auth import authenticate
 
+class getUser(APIView):
+    permission_classes = (IsAuthenticated,)
 
-@api_view(['GET'])
-def getUser(request, pk):
-    user = FiboUser.objects.get(id=pk)
-    serializer = FiboUserSerializer(user, many=False)
-    return Response(serializer.data, status=status.HTTP_200_OK)
+    def get(self, request):
+        usermail = request.user.username
+        user = FiboUser.objects.get(email=usermail)
+        serializer = FiboUserSerializer(user, many=False)
+        return Response(serializer.data, status=status.HTTP_200_OK)
 
 
-@api_view(['DELETE'])
-def deleteUser(request, pk):
-    user = FiboUser.objects.get(id=pk)
-    user.delete()
-    return Response(f'User with id {pk} was deleted', status=status.HTTP_200_OK)
+class deleteUser(APIView):
+    permission_classes = (IsAuthenticated,)
 
+    def delete(self, request):
+        usermail = request.user.username
+        user = FiboUser.objects.get(email=usermail)
+        user.delete()
+        return Response(f'Your user account was deleted', status=status.HTTP_200_OK)
 
-@api_view(['POST'])
-def registerUser(request):
-    # Note that username and email are the same
-    user = FiboUser.objects.create_user(username=request.data['email'], email=request.data['email'],
-                                        password=request.data['password'])
-    return Response(f'User with id {user.id} and email {user.email} was created', status=status.HTTP_201_CREATED)
 
+class registerUser(APIView):
+    permission_classes = (AllowAny, )
```

Comment:
```suggestion
class RegisterUser(APIView):
    permission_classes = (AllowAny, )
```

Please use CamelCase for classes and reformat the file so that the underlinings disappear :)

Reply:
made the changes